agent/connect/ca/provider_vault_test.go

```diff
@@ -356,9 +398,18 @@ func testVaultProviderWithConfig(t *testing.T, isPrimary bool, rawConf map[strin
 
 	testVault.WaitUntilReady(t)
 
+	provider, err := createVaultProvider(isPrimary, testVault.addr, testVault.rootToken, rawConf)
+	if err != nil {
+		testVault.Stop()
+		t.Fatalf("err: %v", err)
+	}
+	return provider, testVault
+}
+
+func createVaultProvider(isPrimary bool, addr, token string, rawConf map[string]interface{}) (*VaultProvider, error) {
```

Comment:
Since this is a test helper, adding `t.Cleanup(provider.Shutdown)` should ensure that all tests properly cleanup.

If it accepts a `testing.T` it could continue to use assertions instead of returning errors, as long as it has a `t.Helper()`.